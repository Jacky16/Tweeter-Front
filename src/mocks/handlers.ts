// src/mocks/handlers.js
import { rest } from "msw";
import requestsUrl from "../config/requestsUrl";

export const handlers = [
  rest.post(requestsUrl.registerUser, (req, res, ctx) => {
    return res(ctx.status(201), ctx.json({}));
  }),
];
