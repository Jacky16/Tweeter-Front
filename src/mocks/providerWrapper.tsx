import { ThemeProvider } from "@mui/system";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { store } from "../redux/store";
import mainTheme from "../styles/mainTheme";

interface ProviderWrapperProps {
  children: JSX.Element | JSX.Element[];
}
const ProviderWrapper = ({ children }: ProviderWrapperProps) => {
  return (
    <BrowserRouter>
      <ThemeProvider theme={mainTheme}>
        <Provider store={store}>{children}</Provider>
      </ThemeProvider>
    </BrowserRouter>
  );
};

export default ProviderWrapper;
