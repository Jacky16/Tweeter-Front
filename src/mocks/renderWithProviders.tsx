import { configureStore, PreloadedState } from "@reduxjs/toolkit";
import { render, RenderOptions } from "@testing-library/react";
import { PropsWithChildren } from "react";
import { Provider } from "react-redux";
import { BrowserRouter, MemoryRouter } from "react-router-dom";
import { RootState, store } from "../redux/store";
import { tweetsReducer } from "../redux/tweetsSlice/tweetsSlice";
import { uiReducer } from "../redux/UiSlice/UiSlice";
import { userReducer } from "../redux/userSlice/userSlice";
import { InitialEntry } from "@remix-run/router";
interface ExtendedPropsWithChildren extends PropsWithChildren {
  initialEntries?: InitialEntry[];
}
interface ExtendedRenderOptions
  extends Omit<RenderOptions, "queries">,
    ExtendedPropsWithChildren {
  preloadedState?: PreloadedState<RootState>;
  store?: typeof store;
}

const Router = ({
  children,
  initialEntries,
}: ExtendedPropsWithChildren): JSX.Element => {
  return initialEntries ? (
    <MemoryRouter initialEntries={initialEntries}>{children}</MemoryRouter>
  ) : (
    <BrowserRouter>{children}</BrowserRouter>
  );
};

export const renderWithProviders = (
  ui: React.ReactElement,
  {
    initialEntries,
    preloadedState,
    store = configureStore({
      reducer: { ui: uiReducer, user: userReducer, tweets: tweetsReducer },
    }),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) => {
  const Wrapper = ({ children }: PropsWithChildren<{}>): JSX.Element => {
    return (
      <Router initialEntries={initialEntries}>
        <Provider store={store}>{children}</Provider>
      </Router>
    );
  };
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};
