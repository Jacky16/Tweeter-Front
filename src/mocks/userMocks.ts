import { UserState } from "../redux/types";

export const mockUser: UserState = {
  isLogged: false,
  token: "abc",
  id: "123",
  alias: "@mario",
  username: "mario",
};
