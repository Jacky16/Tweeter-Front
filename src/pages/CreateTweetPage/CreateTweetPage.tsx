import Grid from "@mui/material/Grid/Grid";
import FormTweet from "../../components/FormCreateTweet/FormCreateTweet";

const CreateTweetPage = () => {
  return (
    <Grid container>
      <Grid item xs={12}>
        <FormTweet />
      </Grid>
    </Grid>
  );
};

export default CreateTweetPage;
