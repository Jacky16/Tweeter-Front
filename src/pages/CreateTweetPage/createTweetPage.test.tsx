import { render } from "@testing-library/react";
import { renderWithProviders } from "../../mocks/renderWithProviders";
import CreateTweetPage from "./CreateTweetPage";

describe("Given the CreateTweetPage component", () => {
  describe("When it is rendered", () => {
    test("Then should match the snapshot", () => {
      const { container } = renderWithProviders(<CreateTweetPage />);
      expect(container).toMatchSnapshot();
    });
  });
});
