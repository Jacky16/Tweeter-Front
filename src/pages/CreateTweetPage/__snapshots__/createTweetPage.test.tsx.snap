// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the CreateTweetPage component When it is rendered Then should match the snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <h1
            class="MuiTypography-root MuiTypography-h1 css-8f85gv-MuiTypography-root"
          >
            Tweet something
          </h1>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-i0bk1x-MuiPaper-root"
          >
            <form>
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 css-1aj15lt-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 css-spompt-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-1 css-1ux21pi-MuiGrid-root"
                  >
                    <div
                      class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
                        data-testid="PersonIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-sm-11 css-6xj14a-MuiGrid-root"
                  >
                    <div
                      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                    >
                      <div
                        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth Mui-focused MuiInputBase-formControl MuiInputBase-multiline css-g9cew8-MuiInputBase-root-MuiInput-root"
                      >
                        <textarea
                          aria-invalid="false"
                          class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline css-66dh3a-MuiInputBase-input-MuiInput-input"
                          id=":r0:"
                          name="description"
                          placeholder="What's happening?"
                          style="height: 0px; overflow: hidden;"
                        />
                        <textarea
                          aria-hidden="true"
                          class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline css-66dh3a-MuiInputBase-input-MuiInput-input"
                          readonly=""
                          style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
                          tabindex="-1"
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-1 MuiGrid-grid-xs-12 css-1a8lh9i-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <label
                      aria-label="Upload image"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                      role="button"
                      tabindex="0"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="ImageIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M21 19V5c0-1.1-.9-2-2-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2zM8.5 13.5l2.5 3.01L14.5 12l4.5 6H5l3.5-4.5z"
                        />
                      </svg>
                      <input
                        hidden=""
                        name="image"
                        type="file"
                      />
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </label>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-sizeSmall  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
                    >
                      <div
                        aria-expanded="false"
                        aria-haspopup="listbox"
                        aria-labelledby="mui-component-select-category"
                        class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-jedpe8-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                        id="mui-component-select-category"
                        role="button"
                        tabindex="0"
                      >
                        Comedy
                      </div>
                      <input
                        aria-hidden="true"
                        class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                        name="category"
                        tabindex="-1"
                        value="comedy"
                      />
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                        data-testid="ArrowDropDownIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M7 10l5 5 5-5z"
                        />
                      </svg>
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-ihdtdm"
                        >
                          <span
                            class="notranslate"
                          >
                            â€‹
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-sizeSmall  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
                    >
                      <div
                        aria-expanded="false"
                        aria-haspopup="listbox"
                        aria-labelledby="mui-component-select-visibilityOpen"
                        class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-jedpe8-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                        id="mui-component-select-visibilityOpen"
                        role="button"
                        tabindex="0"
                      >
                        Everyone
                      </div>
                      <input
                        aria-hidden="true"
                        class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                        name="visibilityOpen"
                        tabindex="-1"
                        value="true"
                      />
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                        data-testid="ArrowDropDownIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M7 10l5 5 5-5z"
                        />
                      </svg>
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-ihdtdm"
                        >
                          <span
                            class="notranslate"
                          >
                            â€‹
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <span
                      aria-valuenow="0"
                      class="MuiCircularProgress-root MuiCircularProgress-determinate MuiCircularProgress-colorPrimary css-1rn30mb-MuiCircularProgress-root"
                      role="progressbar"
                      style="width: 40px; height: 40px; transform: rotate(-90deg);"
                    >
                      <svg
                        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                        viewBox="22 22 44 44"
                      >
                        <circle
                          class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate css-oxts8u-MuiCircularProgress-circle"
                          cx="44"
                          cy="44"
                          fill="none"
                          r="20.2"
                          stroke-width="3.6"
                          style="stroke-dasharray: 126.920; stroke-dashoffset: 126.920px;"
                        />
                      </svg>
                    </span>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-4 MuiGrid-grid-xs-12 css-7lrgoq-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge css-l0xfio-MuiButtonBase-root-MuiButton-root"
                      tabindex="0"
                      type="button"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body1 css-vjoc2v-MuiTypography-root"
                      >
                        Cancel
                      </p>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-fullWidth css-1hwudko-MuiButtonBase-root-MuiButton-root"
                      tabindex="0"
                      type="submit"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body1 css-vjoc2v-MuiTypography-root"
                      >
                        Tweet
                      </p>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
